py_v_sdk/contract/atomic_swap_ctrt.py

```diff
@@ -162,3 +184,230 @@ async def get_tok_bal(self, addr: str) -> int:
+                ctrt_id=self._ctrt_id,
+                func_id=self.FuncIdx.LOCK,
+                data_stack=de.DataStack(
+                    de.Amount.for_tok_amount(amount, unit),
+                    de.Addr(md.Addr(recipient)),
+                    de.Bytes(md.Bytes(puzzle_bytes)),
+                    de.Timestamp(md.VSYSTimestamp(expire_time * SCALE)),
+                ),
+                timestamp=md.VSYSTimestamp.now(),
+                attachment=md.Str(attachment),
+                fee=md.ExecCtrtFee(fee),
+            )
+        )
+        logger.debug(data)
+        return data
+
+    async def maker_solve(
+        self,
+        by: acnt.Account,
+        taker_ctrt_id: str,
+        tx_id: str,
+        key: str,
+        attachment: str = "",
+        fee: int = md.ExecCtrtFee.DEFAULT,
+    ) -> Dict[str, Any]:
+        """
+        maker_solve encapsulates the secret.
+
+        Args:
+            by (acnt.Account): The action maker.
+            taker_ctrt_id (str): The swap ctrt id of the taker's.
+            tx_id (str): The lock transaction id of taker's .
+            key (str): The secret.
+            attachment (str, optional): The attachment of this action. Defaults to "".
+            fee (int, optional): Execution fee of this tx. Defaults to md.ExecCtrtFee.DEFAULT.
+
+        Returns:
+            Dict[str, Any]: The response returned by the Node API.
+        """
+
+        data = await by._execute_contract(
+            tx.ExecCtrtFuncTxReq(
+                ctrt_id=md.CtrtID(taker_ctrt_id),
+                func_id=self.FuncIdx.SOLVE_PUZZLE,
+                data_stack=de.DataStack(
+                    de.Bytes(md.Bytes(base58.b58decode(tx_id))),
+                    de.Bytes(md.Bytes(key.encode("latin-1"))),
+                ),
+                timestamp=md.VSYSTimestamp.now(),
+                attachment=md.Str(attachment),
+                fee=md.ExecCtrtFee(fee),
+            )
+        )
+        logger.debug(data)
+        return data
+
+    async def taker_solve(
+        self,
+        by: acnt.Account,
+        maker_ctrt_id: str,
+        maker_lock_tx_id: str,
+        maker_solve_tx_id: str,
+        attachment: str = "",
+        fee: int = md.ExecCtrtFee.DEFAULT,
+    ) -> Dict[str, Any]:
+        """
+        Taker_solve gets the puzzle.
+
+        Args:
+            by (acnt.Account): The action maker.
+            maker_ctrt_id (str): The contract id of the maker's.
+            maker_lock_tx_id (str): The lock tx id of the maker's.
+            maker_solve_tx_id (str): The solve tx id of the maker's.
+            attachment (str, optional): The attachment of this action. Defaults to "".
+            fee (int, optional): Execution fee of this tx. Defaults to md.ExecCtrtFee.DEFAULT.
+
+        Returns:
+            Dict[str, Any]: The response returned by the Node API
+        """
+        # get the revealed_secret
+        dict_data = await by.chain.api.tx.get_info(maker_solve_tx_id)
```

Comment:
As a good practice, it is actually okay to use just `d` or `data` as the variable name as the scope is limited to the function and the context is quite clear. Verbose name might harm readability as well.